redfish-core/lib/pcie_slots.hpp

```diff
@@ -90,6 +90,55 @@ inline std::string analysisSlotType(const std::string& slotType)
     return "";
 }
 
+inline void
+    getLinkedPcieDevice(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& slotPath, size_t index)
+{
+    // Collect device associated with this slot and
+    // populate it here
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, index](
+            const boost::system::error_code ec,
+            const std::vector<std::pair<
+                std::string,
+                std::vector<std::pair<std::string, std::vector<std::string>>>>>&
+                subtree) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "D-Bus response error on GetSubTree " << ec;
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            if (subtree.size() == 0)
+            {
+                BMCWEB_LOG_ERROR
+                    << "Can't find PCIeDevice D-Bus object for given slot";
+                return;
+            }
+
+            // There should be only one path under the slot.
+            const std::string& pcieDevciePath = std::get<0>(subtree[0]);
+            const std::string pcieDevice =
+                sdbusplus::message::object_path(pcieDevciePath).filename();
```

Comment:
same question here as in the other patch in this PR about use of this upstream.  would this be allowed?  it seems inefficient given that we are starting with a string....

Reply:
Same as above. There was a comment to use wrapper apis as far as possible for any operation on path.

Reply:
👍